apollo-federation/src/supergraph/mod.rs

```diff
@@ -2226,14 +2226,56 @@ fn extract_join_directives(
         }
     }
 
+    for intf_field_pos in &join_directives.interface_fields {
```

Comment:
This is the _exact_ file which caused the `rustc` bug before, so please move this new content into a new file 😁

Reply:
good point!